fmriprep/interfaces/cifti.py

```diff
@@ -0,0 +1,219 @@
+                                 usedefault=True, desc="CIFTI surface target space")
+    subjects_dir = Directory(mandatory=True, desc="FreeSurfer SUBJECTS_DIR")
+    TR = traits.Float(mandatory=True, desc="repetition time")
+    gifti_files = traits.List(File(exists=True), mandatory=True,
+                              desc="surface geometry files")
+
+
+class GenerateCiftiOutputSpec(TraitedSpec):
+    out_file = File(exists=True, desc="generated CIFTI file")
+
+
+class GenerateCifti(SimpleInterface):
+    """
+    Generate CIFTI image from BOLD file in target space
+    """
+    input_spec = GenerateCiftiInputSpec
+    output_spec = GenerateCiftiOutputSpec
+
+    def _run_interface(self, runtime):
+        self.annotation_files, self.label_file = self._fetch_data()
+        self._results["out_file"] = create_cifti_image(
+            self.bold_file,
+            self.label_file,
+            self.annotation_files,
+            self.gifti_files,
+            self.volume_target,
+            self.surface_target,
+            self.TR)
+
+        return runtime
+
+    def _fetch_data(self):
+        """Converts inputspec to files"""
+        if self.surface_target == "fsnative":
+            raise NotImplementedError
+
+        annotation_files = sorted(glob(os.path.join(self.subjects_dir,
+                                                    self.surface_target,
+                                                    'label',
+                                                    '*h.aparc.annot')))
+        # TODO: fetch label_file
+        return annotation_files
+
+
+def create_cifti_image(bold_file, label_file, annotation_files,
+                       gii_files, volume_target, surface_target, tr):
+    """
+    Generate CIFTI image in target space
+    """
+
+    # grab image information
+    bold_img = nb.load(bold_file)
+    bold_data = bold_img.get_data()
+    timepoints = bold_img.shape[3]
+    label_data = nb.load(label_file).get_data()
+
+    # set up CIFTI information
+    model_type = "CIFTI_MODEL_TYPE_VOXELS"
+    series_map = ci.Cifti2MatrixIndicesMap((0, ),
+                                           'CIFTI_INDEX_TYPE_SERIES',
+                                           number_of_series_points=timepoints,
+                                           series_exponent=0,
+                                           series_start=0.0,
+                                           series_step=tr,
+                                           series_unit='SECOND')
+    # Create CIFTI brain models
+    idx_offset = 0
+    brainmodels = []
+    bm_ts = np.empty((timepoints, 0))
+
+    for structure, labels in CIFTI_STRUCT_WITH_LABELS.items():
+        if not labels:  # surface model
+            model_type = "CIFTI_MODEL_TYPE_SURFACE"
+            # use the corresponding annotation
+            hemi = structure.split('_')[-1][0]
+            annots = [fl for fl in annotation_files if ".aparc.annot" in fl]
```

Comment:
While probably fine, this could be made a little more precise:

```Python
annots = [fl for fl in annotation_files if fl.endswith('.aparc.annot')]
```